src/main/java/net/minecraftforge/event/level/LevelEvent.java

```diff
@@ -117,4 +124,82 @@ public ServerLevelData getSettings()
             return settings;
         }
     }
+
+    /**
+     * Called by NaturalSpawner to gather a list of all possible entities that can spawn at the specified location.
+     * If an entry is added to the list, it needs to be a globally unique instance.
+     * The event is called in {@link net.minecraft.world.level.NaturalSpawner#getRandomSpawnMobAt(ServerLevel,
+     * StructureManager, ChunkGenerator, MobCategory, RandomSource, BlockPos)}.
+     * Canceling the event will result in an empty list, meaning no entity will be spawned.
+     */
+    @Cancelable
+    public static class PotentialSpawns extends LevelEvent
+    {
+        private final MobCategory mobcategory;
+        private final BlockPos pos;
+        private final List<MobSpawnSettings.SpawnerData> list;
+
+        public PotentialSpawns(LevelAccessor level, MobCategory category, BlockPos pos, WeightedRandomList<MobSpawnSettings.SpawnerData> oldList)
+        {
+            super(level);
+            this.pos = pos;
+            this.mobcategory = category;
+            if (!oldList.isEmpty())
+                this.list = new ArrayList<>(oldList.unwrap());
+            else
+                this.list = new ArrayList<>();
+        }
+
+        /**
+         * Returns the category of the mobs in the spawn list.
+         *
+         * @return the category of the mobs in the spawn list.
+         */
+        public MobCategory getMobCategory()
+        {
+            return mobcategory;
+        }
+
+        /**
+         * Returns the block position where the chosen mob will be spawned.
+         *
+         * @return the block position where the chosen mob will be spawned.
+         */
+        public BlockPos getPos()
+        {
+            return pos;
+        }
+
+        /**
+         * Returns the list ofmobs that can potentially be spawned.
+         *
+         * @return the list ofmobs that can potentially be spawned.
+         */
+        public WeightedRandomList<MobSpawnSettings.SpawnerData> getSpawnerDataList()
+        {
+            return WeightedRandomList.create(list);
```

Comment:
This feels needlessly complex. just add another field: `private final List<MobSpawnSettings.SpawnerData> view` and initialize it to `Collections.unmodifiableList(list)` and then just make this method return a list instead of a weighted random list. Lists are easier to interact with anyway

Reply:
Ok let me know if the latest push is what you were thinking, or whether I have totally missed the point. :)

Reply:
Yeah that is what I was thinking in regards to this method.